CRM/Utils/Token.php

```diff
@@ -1252,16 +1252,11 @@ public static function getTokenDetails(
         if (CRM_Utils_Array::value('preferred_communication_method', $returnProperties) == 1
           && array_key_exists('preferred_communication_method', $contactDetails[$contactID])
         ) {
-          $pcm = CRM_Core_PseudoConstant::get('CRM_Contact_DAO_Contact', 'preferred_communication_method');
-
           // communication Preference
-          $contactPcm = explode(CRM_Core_DAO::VALUE_SEPARATOR,
-            $contactDetails[$contactID]['preferred_communication_method']
-          );
           $result = array();
-          foreach ($contactPcm as $key => $val) {
+          foreach ($contactDetails[$contactID]['preferred_communication_method'] as $key => $val) {
```

Comment:
@eileenmcnaughton I think you still need the explode here, Lots of failed tests which look like issue here